test/lib/runtime/panel_survey_test.exs

```diff
@@ -0,0 +1,217 @@
+defmodule Ask.Runtime.PanelSurveyTest do
+  use Ask.ModelCase
+  use Ask.TestHelpers
+  alias Ask.Runtime.{PanelSurvey, RespondentGroupAction}
+  alias Ask.{Survey, Repo, TestChannel, Respondent}
+
+  describe "new occurence" do
+    test "creates a new ready occurrence" do
+      panel_survey = completed_panel_survey()
+
+      {result, data} = PanelSurvey.new_occurrence(panel_survey)
+
+      assert result == :ok
+      new_occurrence = Map.get(data, :new_occurrence)
+      assert new_occurrence
+      assert new_occurrence.state == "ready"
+      assert new_occurrence.panel_survey_id == panel_survey.id
+    end
+
+    test "preserves the incentives enabled flag" do
+      panel_survey = incentives_enabled_panel_survey()
+
+      {:ok, %{new_occurrence: new_occurrence}} = PanelSurvey.new_occurrence(panel_survey)
+
+      assert_incentives_enabled(new_occurrence)
+
+      panel_survey = incentives_disabled_panel_survey()
+
+      {:ok, %{new_occurrence: new_occurrence}} = PanelSurvey.new_occurrence(panel_survey)
+
+      assert_incentives_disabled(new_occurrence)
+    end
+
+    test "removes start_date and end_date of the schedule" do
+      panel_survey = scheduled_panel_survey()
+      schedule = Ask.PanelSurvey.latest_occurrence(panel_survey).schedule
+
+      {:ok, %{new_occurrence: new_occurrence}} = PanelSurvey.new_occurrence(panel_survey)
+
+      assert new_occurrence.schedule ==
+        clean_dates(schedule)
+    end
+
+    test "errors when the latest ocurrence isn't terminated" do
+      panel_survey = panel_survey_with_occurrence()
+
+      {result, data} = PanelSurvey.new_occurrence(panel_survey)
+
+      assert result == :error
+      assert Map.get(data, :error) == "Last panel survey occurrence isn't terminated"
+    end
+
+    test "the new occurence is based on the latest occurrence" do
+      panel_survey = completed_panel_survey()
+      previous_occurrence = Ask.PanelSurvey.latest_occurrence(panel_survey)
+      latest_occurrence = insert(:survey, panel_survey: panel_survey, state: "terminated")
+
+      {:ok, %{new_occurrence: new_occurrence}} = PanelSurvey.new_occurrence(panel_survey)
+
+      # TODO: Don't compare the name, compare something more meaningfull
```

Comment:
pending TODO. also, the `refute` and `assert` aren't straightforward to understand why

Reply:
Please let's talk about this test on a call 🙏

Reply:
So, after what we talked in the call the other day, this test should assert that the channel is the same, the respondents are the same, also the schedule (and so on.., right?)

Reply:
Yes! This is pending. Thanks for remembering me it!